src/components/icons/LottieAnimation.tsx

```diff
@@ -35,15 +164,26 @@ const LottieAnimation: React.FC<LottieAnimationProps> = ({
   }
 
   useEffect(() => {
-    if (lottieRef.current) {
+    if (lottieRef.current && animationDataWithColor) {
       lottieRef.current.setSpeed(speed)
+      setKey(prevKey => prevKey + 1) // Forces the re-render to apply the color change
     }
-  }, [speed])
+  }, [speed, animationDataWithColor])
+
+  if (!animationDataWithColor) {
+    console.warn('animationData is not provided.')
```

Comment:
Instead of `console.warn`, we likely either want to fully allow `null` (by not throwing a warning) or fully disallow `null` (by not allowing it in the type).